pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentMessageExpiryMonitor.java

```diff
@@ -231,6 +231,10 @@ public void markDeleteFailed(ManagedLedgerException exception, Object ctx) {
     @Override
     public void findEntryComplete(Position position, Object ctx) {
         if (position != null) {
+            var markDeletedPosition = cursor.getMarkDeletedPosition();
+            if (markDeletedPosition != null && markDeletedPosition.compareTo(position) >= 0) {
```

Comment:
@poorbarcode This seems to miss `expirationCheckInProgress = FALSE;`. Could you please check?

Reply:
The variable will be finally set to `false` after the task is completely finished(after mark deleting task is finished), see also this two lines:
- https://github.com/apache/pulsar/blob/v4.1.1/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentMessageExpiryMonitor.java#L217
- https://github.com/apache/pulsar/blob/v4.1.1/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentMessageExpiryMonitor.java#L226

Reply:
> The variable will be finally set to after the task is completely finished(after mark deleting task is finished), see also this two lines:

Please check again. `markDeleteCallback` won't get executed at all after the new code in this PR has been added.

Reply:
I created #24941 to fix the bug and it also contains a test case

Reply:
Ah, Thanks